altamisa/isatab/models_validation.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+"""Validation of an ISA model or single objects
+
+Eventually, all format independent content- and specification-related validations which
+don't interrupt model creation definitely (e.g. when parsing from ISA-tab) should go
+here. Then, validations can be performed on whole models (e.g. after parsing or before
+writing) and provide a comprehensive list of warnings of different degree.
+"""
+
+from typing import Dict
+import warnings
+
+from ..exceptions import ParseIsatabException, IsaValidationWarning
+from . import headers
+from . import models
+
+
+__author__ = "Mathias Kuhring <mathias.kuhring@bihealth.de>"
+
+
+# Assay and protocol restrictions regarding data, names or annotations -----------------
+
+
+# Assay measurement types (only the once needed for special validations)
```

Comment:
Is this the right place for the constants? Should this go to some models file instead?

Reply:
So far, these constants are only used in validation. However, I will need other constants from other files in different contexts soon (e.g. all the section names in parse_investigation.py). So should we basically pull all of ISA specific constants (sections, headers, restrictions, etc.) into one collective file? Or do you think is to much?

Reply:
What about creating a `constants` module:

```
constants/
  submod1.py
  submod2.py
  __init__.py
```
In `constants/__init__.py` you can then do the following to get them all as `constants.X` instead of `constants.submod1.X`.

```
from .submod1 import *
from .submod2 import *
```